runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java

```diff
@@ -163,6 +163,7 @@ public void onWidgetLoaded(WidgetLoaded event)
 		}
 
 		loadedCombatStyleWidget = true;
+		processWidgets();
```

Comment:
so... is this necessary or not?

Reply:
I didn't think it was necessary up until now, but if you use your f-keys to navigate to the combat tab the moment you log in you will see the disabled options for a split second before the varbits are set.

Reply:
Is the widget hidden change event still necessary if we are now processing widgets when the var changes?

Reply:
Yes, without it the entire plugin breaks. The attack options seem to be living a life of their own.

Reply:
In what cases can the visibility of the combat widgets change when the vars do not?

Reply:
In my experience, it looks as if the attack style widgets reset their visibility whenever you go to the combat tab. So you have to set it to hidden before they are properly loaded.

Reply:
It looks like there is a hidden even fired when the tab is selected and when it is deselected. `loadedCombatStyleWidget` will get reset whenever you teleport so that will break the plugin

Reply:
I seem to have misinterpreted that event. Thanks, I think it's fixed now.